### PR TITLE
Enforce POST in form validation

### PR DESCRIPTION
**Pull request by sarah-witt**
### What does this PR do?

Enforces post requests in form validation.

### Additional Notes

Following:
https://www.jenkins.io/doc/developer/security/form-validation/

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java [hunk 1]
```diff
@@ -279,6 +284,8 @@ private boolean validateTargetHost(String targetHost) {
      * @return a FormValidation object used to display a message to the user on the configuration
      * screen.
      */
+
+    @POST
```
Comment (FlorianVeaux) [inline src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java:288]:
```suggestion
     */
    @POST
```

File: src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java [hunk 2]
```diff
@@ -321,6 +329,8 @@ private boolean validateTargetLogCollectionPort(String targetLogCollectionPort)
...
```
Comment (FlorianVeaux) [inline src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java:333]:
```suggestion
     */
    @POST
```

**Comment by FlorianVeaux** [comment]
@sarah-witt Is that ready to be merged ? Can we rebase first also? I don't think `POST` exists with our current dependencies, we may have to use `RequirePOST` instead

**Comment by sarah-witt** [comment]
Sure, I'll update it!

**Comment by sarah-witt** [comment]
Updated and tested!

**Comment by FlorianVeaux** [comment]
:shipit:

**Pull request merged**